spring-rabbit/src/main/java/org/springframework/amqp/rabbit/connection/CachingConnectionFactory.java

```diff
@@ -882,4 +892,48 @@ public String toString() {
 
 	}
 
+	/**
+	 * A strategy interface for conditional exception logging.
+	 * @since 1.5
+	 *
+	 */
+	public interface ConditionalExceptionLogger {
+
+		void log(Log logger, Throwable t);
+
+	}
+
+	/**
+	 * Default implementation of {@link ConditionalExceptionLogger} for logging channel
+	 * close exceptions.
+	 * @since 1.5
+	 *
+	 */
+	private class DefaultChannelCloseLogger implements ConditionalExceptionLogger {
```

Comment:
Yes, that's good catch. Having this particular context it our responsibility to emit the proper message on the matter, but that is up to end-user how to log in his implementation or even do something else.
If he isn't interested in the our message, he always can ignore it.